GDJS/scripts/generate-esm-types.js

```diff
@@ -0,0 +1,30 @@
+const {
+  compileESModuleTypeDefinitions,
+} = require('./lib/build-esm-library-types');
+const { getAllInOutFilePaths } = require('./lib/runtime-files-list');
+
+(async () => {
+  // Generate the output file paths
+  const { allESMInOutFilePaths } = await getAllInOutFilePaths({
+    bundledOutPath: __dirname, // Dummy directory name, since we do not care about compilation destination.
+  });
+
+  const entryPointsForNamespace = new Map();
+  for (const { inPath, namespace } of allESMInOutFilePaths) {
+    const entryPoints = entryPointsForNamespace.get(namespace) || [];
+    entryPoints.push(inPath);
+    entryPointsForNamespace.set(namespace, entryPoints);
+  }
+
+  const namespacesWithEntryPoints = Array.from(
+    entryPointsForNamespace.entries()
+  );
+
+  await Promise.all(
+    namespacesWithEntryPoints.map(([namespace, entryPoints]) =>
```

Comment:
Why not inline `namespacesWithEntryPoints`?
Do you think `for await ( of )` could be used to iterate on `entryPointsForNamespace.entries()` directly?

Reply:
I thought that makes the code a bit easier to read, with the name of the variable giving an indication of what that step is supposed to do.

Reply:
It is just an `Array` representation of a `Map`. The 2 variables have the same meaning.
Adding a variable declaration make it looks as if there were something important going on.
Moreover, it's another variable to jump from to navigate the code.

Reply:
Ah I understand now, I'll change this 👍